pyrsa/data/dataset.py

```diff
@@ -349,3 +349,126 @@ def dataset_from_dict(data_dict):
     else:
         raise ValueError('type of Dataset not recognized')
     return data
+
+
+def merge_subsets(dataset_list):
+    """
+    Generate a dataset object from a list of smaller dataset objects (e.g., as generated by the subset_* methods).
+    Assumes that descriptors, channel descriptors and number of channels per observation match.
+
+    Args:
+        dataset_list (list):
+            List containing PyRSA datasets
+    
+    Returns:
+        merged_dataset (Dataset):
+            PyRSA dataset created from all datasets in dataset_list
+    """
+    assert isinstance(dataset_list, list), "Provided object is not a list."
+    assert "Dataset" in str(type(dataset_list[0])), "Provided list does not only contain Dataset objects."
+    
+    baseline_ds = dataset_list[0]
+    descriptors = baseline_ds.descriptors.copy()
+    channel_descriptors = baseline_ds.channel_descriptors.copy()
+    measurements = baseline_ds.measurements.copy()
+    obs_descriptors = baseline_ds.obs_descriptors.copy()
+    
+    for ds in dataset_list[1:]:
+        assert "Dataset" in str(type(ds)), "Provided list does not only contain Dataset objects."
+        assert descriptors == ds.descriptors.copy(), "Dataset descriptors do not match."
+        measurements = np.append(measurements, ds.measurements, axis=0)
+        obs_descriptors = append_obs_descriptors(obs_descriptors, ds.obs_descriptors.copy())
+    
+    merged_dataset = Dataset(measurements,
+                                descriptors = descriptors, 
+                                obs_descriptors = obs_descriptors,                             
+                                channel_descriptors = channel_descriptors)
+    return merged_dataset
+
+
+def append_obs_descriptors(dict_orig, dict_addit):
+    """
+    Merge two dictionaries of observation descriptors with matching keys and numpy arrays as values.
+    """
+    assert list(dict_orig.keys()) == list(dict_addit.keys()), "Provided observation descriptors have different keys."
+    
+    dict_merged = {}
+    keys = list(dict_orig.keys())
+    for k in keys:
+        values = np.array(np.append(dict_orig[k], dict_addit[k]))
+        dict_merged.update({k:values})
+    return dict_merged
+
+def odd_even_split(dataset, obs_desc, sort_by = None):
```

Comment:
I think this one can probably be done more efficiently by splitting the descriptor list and running two subset operations on the original dataset?

Reply:
Also: Do we need the sort_by option? I mean you could just run the sort command afterwards, right?

Reply:
I‘m not sure what you mean by running two subset operations on the original dataset. Could you please express this in pseudocode?

Reply:
I find it handy but I wouldn‘t mind removing the sort_by option.

Reply:
Sure, it would be something like this:
     even_values =  np.unique(dataset.obs_descriptors[obs_desc])[0::2]
     odd_values = np.unique(dataset.obs_descriptors[obs_desc])[1::2]
even_split = dataset.subset(obs_desc, even_values)
odd_split = dataset.subset(obs_desc, odd_values)